### ISSUE TITLE
updates available notification

### ISSUE THREAD

**Issue by fibo** [fisherman/fisherman#65]
To enhance user experience, it would be nice to have a message that notifies when fisherman updates are available, a la [bower](http://bower.io/).

This could be displayed after an install.
It should be silenced if some `--quiet` option is enabled, to respet [rule of silence](https://en.wikipedia.org/wiki/Unix_philosophy).

**Comment by bucaran** [comment]
Just to be clear, you want this notification to show up without user intervention?

What about:

```
fisher update --check
```

or 

```
fisher --check-for-updates
```

**Comment by fibo** [comment]
I think [bower](http://bower.io/) behaviour is the right way.
Users will not launch a command to check for updates, usually: notification should be displayed after you perform an `fisher install foo`, which needs to connect to the network, so you can also try to get latest fisher version and compare it to the local one.

If user want to update fisher, could launch a command like `fisher upgrade`.

Instead, `fisher update` usually it is intended to be used to update installed packages.

Hopes this helps to clarify and define better this feature request, which is something many other advanced package managers has and makes users feel more comfortable.

**Comment by bucaran** [comment]
Commands that could trigger a notification are install and update, maybe search.

I think the right approach is this, but I would like to limit the update checks to once a day as is unlikely Fisherman will be updated that often.

The index, which is the database of packages is self-updating so this feature would only apply to Fisherman the manager itself.

**Comment by kiRach** [comment]
Maybe also consider Oh My Zsh behaviour here? I was pretty happy when used it It checks for updates automatically every week or so (customisable) when you launch a shell.

**Comment by bucaran** [comment]
>  checks for updates automatically every week or so (customisable) when you launch a shell.

That makes sense too.

In 0.4.0 the index is updated every time there is a search query, in 0.5.0 there is a new mechanism that updates the index only once per search increasing install/uninstall time of local plugins from 1s to instantaneous.

**Comment by bucaran** [comment]
Closing in favor of #99.

Basically, the update mechanism needs to be seriously updated. I expect to add all major improvements for the next milestone.
